### PR TITLE
[Enhancement] Automatically create block cache directories if it is configured but not exist.

### PR DESCRIPTION
**Pull request by GavinMar**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by DorianZheng — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

File: be/src/block_cache/block_cache.cpp [hunk 1]
```diff
@@ -34,7 +35,15 @@ BlockCache* BlockCache::instance() {
 }
 
 Status BlockCache::init(const CacheOptions& options) {
-    // TODO: check block size limit
+    for (auto& dir : options.disk_spaces) {
+        if (dir.size > 0 && !fs::path_exist(dir.path)) {
```
[the remark this answers is not in the document]
Comment (GavinMar) [reply be/src/block_cache/block_cache.cpp]: > What will happen if `dir.path` is a file?

If it is a file, the BE will start failed when initialize underly kv cache. I have update the code to check this case when initializing block cache.

File: be/src/block_cache/block_cache.cpp [hunk 2]
```diff
@@ -34,7 +35,24 @@ BlockCache* BlockCache::instance() {
...
+        if (fs::path_exist(dir.path)) {
+            ASSIGN_OR_RETURN(const bool is_dir, fs::is_directory(dir.path));
+            if (!is_dir) {
+                LOG(ERROR) << "the block cache disk path already exists but not a directory, path=" << dir.path;
+                return Status::InvalidArgument("invalid block cache disk path");
+            }
+        } else {
+            auto create_res = fs::create_directories(dir.path);
+            if (!create_res.ok()) {
+                LOG(ERROR) << "create block cache disk path failed, path=" << dir.path;
```
Comment (imay) [inline be/src/block_cache/block_cache.cpp:51]: better to print the status message.

Comment (GavinMar) [reply be/src/block_cache/block_cache.cpp]: > better to print the status message.

OK

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**